memory-viz/src/memory_model.ts

```diff
@@ -943,9 +947,18 @@ export class MemoryModel {
 
         style = { ...style, ...this.roughjs_config?.options };
 
-        this.svg.appendChild(
-            this.rough_svg.rectangle(x, y, width, height, style)
+        const rectElement = this.rough_svg.rectangle(
+            x,
+            y,
+            width,
+            height,
+            style
         );
+
+        rectElement.setAttribute("id", `object-${this.objectCounter}`);
```

Comment:
Overall this approach is good, but you should add an additional parameter to `drawRect` to indicate whether the rect is the bounding box of an object or not. This is because each object can call `drawRect` multiple times, and we only want the ids to apply to the outer rectangles.